src/test/ai-accuracy-tests/ai-accuracy-tests.ts

```diff
@@ -87,11 +138,69 @@ const testCases: TestCase[] = [
         .find()
         .toArray();
 
-      expect(documents).to.deep.equal(fixtures.CookBook.recipes);
+      expect(documents).to.deep.equal(
+        fixtures.CookBook.recipes.documents.map((doc) => {
+          if (doc.title === 'Beef Wellington') {
+            return {
+              ...doc,
+              preparationTime: 150,
+              difficulty: 'Very Hard',
+            };
+          }
+          return doc;
+        })
+      );
+    },
+  },
+  {
+    testCase: 'Aggregation with averaging and filtering',
+    type: 'query',
+    databaseName: 'pets',
+    collectionName: 'competition-results',
+    userInput:
+      'What is the average score for dogs competing in the best costume category? Put it in a field called "avgScore"',
+    assertResult: async ({
+      responseContent,
+      connectionString,
+    }: AssertProps) => {
+      const output = await runCodeInMessage(responseContent, connectionString);
+
+      expect(output.data?.result?.content[0]).to.deep.equal({
+        avgScore: 9.3,
+      });
+    },
+  },
+  {
+    testCase: 'Create an index',
+    type: 'query',
+    databaseName: 'FarmData',
+    collectionName: 'Pineapples',
+    reloadFixtureOnEachRun: true,
+    userInput:
+      'How to index the harvested date and sweetness to speed up requests for sweet pineapples harvested after a specific date?',
+    assertResult: async ({
+      responseContent,
+      connectionString,
+      mongoClient,
+    }: AssertProps) => {
+      await runCodeInMessage(responseContent, connectionString);
+
+      const indexes = await mongoClient
+        .db('FarmData')
+        .collection('Pineapples')
+        .listIndexes()
+        .toArray();
+
+      expect(indexes.length).to.equal(2);
```

Comment:
[nit] Should we add an assertion that such indexes don't exist prior to "act" phase? Not familiar with the test harness setup, but whenever we have tests that mutate the data, I like having a sanity assertion prior to the mutation that ensures the original state of the data is different from the final state.

Reply:
I like it! Updated this and another test that could benefit from that.